### PR TITLE
Add TAL_ZH_ADULT_TEACH ASR recipe (ESPnet Bootcamp)

### PR DESCRIPTION
**Pull request by HsunGong**
## What did you change?

This pull request adds a new ESPnet recipe for training automatic speech recognition (ASR) models on the **Tal-zh-adult-teach** dataset.  
The recipe is developed as part of the **WavLab bootcamp project** and includes data preparation, training, and decoding configurations.

---

## Why did you make this change?

The **Tal-zh-adult-teach** dataset, published at  
https://ai.100tal.com/openData/voice,  
contains approximately **100 hours of Mandarin Chinese speech data** collected from online courses taught by professional instructors.

Key characteristics of the dataset:

- Audio consists of **teacher lecture recordings** from online courses
- Covers **Chinese language and mathematics** subjects
- **80+ speakers**, with **one speaker per utterance**
- Transcriptions include **subject labels and speaker IDs**
- Dataset split:
  - Training / Validation / Test = **7 : 1 : 2**
- Total size: **~9.03 GB** (3 files)

This dataset is well-suited for building **real-world educational-domain ASR systems**, and adding an official ESPnet recipe lowers the barrier for reproducible research and future benchmarking.

---

## Is your PR small enough?

Yes.

- only one new recipe added

---

### Pretrained Models

Trained models have been uploaded to Hugging Face:

👉 https://huggingface.co/espnet/xun_tal_zh_adult_teach_branchformer

### Evaluation Results (CER)

| Dataset | Snt | Wrd | Corr | Sub | Del | Ins | Err | S.Err |
|---|---:|---:|---:|---:|---:|---:|---:|---:|
| decode_asr_branchformer_asr_model_valid.acc.ave / test | 6072 | 160057 | 89.4 | 8.2 | 2.4 | 0.9 | 11.4 | 78.6 |
| decode_asr_branchformer + LM / test | 6072 | 160057 | 89.2 | 8.2 | 2.7 | 1.1 | 11.9 | 78.5 |
| decode_asr_branchformer_asr_model_valid.acc.ave / org/dev | 3208 | 77334 | 91.3 | 7.2 | 1.5 | 0.8 | 9.6 | 72.8 |
| decode_asr_branchformer + LM / org/dev | 3208 | 77334 | 90.9 | 7.4 | 1.7 | 1.0 | 10.1 | 72.5 |

### CODE REVIEW

**Comment by sw005320** [comment]
@ftshijt, could you review this PR?

File: egs2/tal_zh_adult_teach/asr1/run.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# speed perturbation related
+# (train_set will be "${train_set}_sp" if speed_perturb_factors is specified)
+speed_perturb_factors="0.9 1.0 1.1"
+
+./asr.sh \
+    --nj 64 \
+    --inference_nj 8 \
+    --gpu_inference true \
+    --ngpu 4 \
```
Comment (ftshijt) [inline egs2/tal_zh_adult_teach/asr1/run.sh:26]: usually, we can set these hyperparameters to default ones

Comment (ftshijt) [reply egs2/tal_zh_adult_teach/asr1/run.sh:26]: (same for others)

**Review by ftshijt — COMMENTED** [review]
Many thanks! The current PR looks very good and well-organized already. I just left a minor comments for style consistency.

Thanks for your contribution!

**Comment by HsunGong** [comment]
@ftshijt I've changed to default params.

**Comment by sw005320** [comment]
LGTM
After the CI test, I'll merge this PR.

**Comment by sw005320** [comment]
Thanks, @HsunGong!
